test/quicapitest.c

```diff
@@ -2863,6 +2863,140 @@ static int test_ssl_set_verify(void)
     return testresult;
 }
 
+/*
+ * Creates a server-initiated stream (unidirectional if is_uni=1, bidirectional
+ * otherwise) and tests that client calling SSL_accept_stream with accept_flags
+ * behaves as expected.
+ */
+static int create_accept_stream(SSL *serverssl, SSL *clientssl,
+                                uint64_t accept_flags, int is_uni)
+{
+    unsigned char buf[16], msg[] = "Hello, World!";
+    SSL *clientstream = NULL, *serverstream = NULL, *stream = NULL;
+    int ret = 0, should_accept = 1;
+    uint64_t new_stream_flags = is_uni ? SSL_STREAM_FLAG_UNI : 0;
+    size_t nread, nwritten;
+
+    if (is_uni != 0 && is_uni != 1)
+        goto err;
+
+    if ((accept_flags & SSL_ACCEPT_STREAM_UNI)
+        && !(accept_flags & SSL_ACCEPT_STREAM_BIDI))
+        should_accept = is_uni;
+    else if ((accept_flags & SSL_ACCEPT_STREAM_BIDI)
+             && !(accept_flags & SSL_ACCEPT_STREAM_UNI))
+        should_accept = !is_uni;
+
+    if (!TEST_ptr(serverstream = SSL_new_stream(serverssl, new_stream_flags))
+        || !TEST_int_gt(SSL_write_ex(serverstream, msg, sizeof(msg), &nwritten), 0)
+        || !TEST_int_eq(nwritten, sizeof(msg))
+        || !TEST_int_eq(SSL_handle_events(clientssl), 1))
+        goto err;
```

Comment:
I think that for the test to properly test that the right stream is accepted, you should create two server-initiated streams, one unidirectional and one bidirectional at once. Or maybe you can do more iterations (i.e. only uni, only bidi, both).

On accept you should test that according to the flags set, the appropriate stream type is accepted using SSL_get_stream_type().